pkg/cloudprovider/providers/openstack/openstack_volumes.go

```diff
@@ -86,73 +252,45 @@ func (os *OpenStack) DetachDisk(instanceID string, partialDiskId string) error {
 		glog.Errorf("Unable to initialize nova client for region: %s", os.region)
 		return err
 	}
-	if len(disk.Attachments) > 0 && disk.Attachments[0]["server_id"] != nil && instanceID == disk.Attachments[0]["server_id"] {
+	if volume.AttachedServerId != instanceID {
+		errMsg := fmt.Sprintf("Disk: %s has no attachments or is not attached to compute: %s", volume.Name, instanceID)
+		glog.Errorf(errMsg)
```

Comment:
minor point: This is an openstack-python pattern, and not usual for golang.  Usually you want to glog the error *xor* return the error to your caller, presumably so they might glog it later with some additional context.